lib/Serialization/Deserialization.cpp

```diff
@@ -6379,6 +6379,11 @@ void ModuleFile::loadAllMembers(Decl *container, uint64_t contextData) {
   }
 }
 
+void ModuleFile::diagnoseMissingNamedMember(const IterableDeclContext *IDC,
+                                            DeclName name) {
+  return;
```

Comment:
Nit: we shouldn't never take this path, right? Should this be an `llvm_unreachable` maybe?

Reply:
I'm not sure about the path never being taken. The call is made here in `NameLookup.cpp`. If `ci->loader` is a `ModuleFile`, we will hit that path. I don't know enough about module files at the moment to know if that happens. Maybe a comment or a TODO explaining that diagnostics haven't been implemeted for module files would suffice? Not sure.

```cpp
 auto known = Table.find(name);
  if (known == Table.end()) {
    if (ctx.LangOpts.EnableExperimentalClangImporterDiagnostics &&
        ctx.isLazyContext(decl)) {
      auto ci =
          ctx.getOrCreateLazyIterableContextData(decl, /*lazyLoader=*/nullptr);
      ci->loader->diagnoseMissingNamedMember(decl, name);
    }
    return TinyPtrVector<ValueDecl *>();
  }
```

Reply:
Hmm I guess that brings up another point: will these errors ever get triggered for Swift decls? For example if someone wrote:

```
struct X {}
let y = X().y
```

(There may be an earlier error that catches that, but you get the point.)

I think we probably want to add a check so we only diagnose clang decls and then we can add an assertion + TODO in the module file overload. Otherwise this might silently do the wrong (or unexpected) thing, which is not ideal.

Reply:
I believe it could get triggered for Swift decls, but obviously the empty overload would do nothing. I've added a check that we are dealing with a ClangDecl and asserted the other overload is not visited.